study-builder/src/main/java/org/broadinstitute/ddp/studybuilder/ActivityBuilder.java

```diff
@@ -333,15 +331,25 @@ public Config readDefinitionConfig(String filepath, boolean allowUnresolved) {
         return definition;
     }
 
+    /**
+     * Build an instance of {@link ActivityDef}.
+     *
+     * <p><b>Steps:</b>
+     * <ul>
+     *   <li>build an instance of {@link ActivityDef} from {@link Config} definition;</li>
+     *   <li>if command-line option `process-translations` is specified then run the process of
+     *       translations' references automatic generation;</li>
+     *   <li>validate the {@link ActivityDef}.</li>
+     * </ul>
+     */
+    private ActivityDef buildActivityDefFromConfig(Config definition) {
+        ActivityDef activityDef = gson.fromJson(ConfigUtil.toJson(definition), ActivityDef.class);
+        activityDefTranslationsProcessor.run(activityDef);
+        validateDefinition(activityDef);
```

Comment:
After the ActivityDef is built the translations processing is called (and it will be really called only if command-line option `process-translations` is specified. In unit tests this command-line flag not affects: the boolean flag set directly in the code.